components/match2/commons/match_group_input.lua

```diff
@@ -24,41 +24,40 @@ local globalVars = PageVariableNamespace({cached = true})
 local MatchGroupInput = {}
 
 function MatchGroupInput.readMatchlist(bracketId, args)
-	local matches = {}
-	for matchKey, matchArgs in Table.iter.pairsByPrefix(args, 'M') do
-		local matchIndex = tonumber(matchKey:match('(%d+)$'))
-		local matchId = string.format('%04d', matchIndex)
+	local matchKeys = Table.mapArgumentsByPrefix(args, {'M'}, FnUtil.identity)
 
-		matchArgs = Json.parse(matchArgs)
+	return Array.map(matchKeys, function(matchKey, matchIndex)
+			local matchId = string.format('%04d', matchIndex)
```

Comment:
Extract this string format so that we can use the function name as explanation of what it's doing